Tests/Data/Parabolic/Richards/infiltration_1D/1d-infiltration.prj

```diff
@@ -0,0 +1,178 @@
+                    <type>AqueousLiquid</type>
+                    <properties>
+                        <property>
+                            <name>viscosity</name>
+                            <type>Constant</type>
+                            <value>1.e-3</value>
+                        </property>
+                        <property>
+                            <name>density</name>
+                            <type>Constant</type>
+                            <value>1.e3</value>
+                        </property>
+                    </properties>
+                </phase>
+                <phase>
+                    <type>Solid</type>
+                    <properties>
+                        <property>
+                            <name>storage</name>
+                            <type>Constant</type>
+                            <value>0</value>
+                        </property>
+                        <property>
+                            <name>porosity</name>
+                            <type>Constant</type>
+                            <value>0.43</value>
+                        </property>
+                        <property>
+                            <name>permeability</name>
+                            <type>Constant</type>
+                            <value>2.94484086533016E-13</value>
+                        </property>
+                    </properties>
+                </phase>
+            </phases>
+            <properties>
+                <property>
+                    <name>reference_temperature</name>
+                    <type>Constant</type>
+                    <value>293.15</value>
+                </property>
+                <property>
+                    <name>relative_permeability</name>
+                    <type>RelativePermeabilityVanGenuchten</type>
+                    <exponent>0.358974358974359</exponent>
+                    <minimum_relative_permeability_liquid>1e-3</minimum_relative_permeability_liquid>
+                    <residual_liquid_saturation>0.18139535</residual_liquid_saturation>
+                    <residual_gas_saturation>0.0</residual_gas_saturation>
+                </property>
+                <property>
+                    <name>saturation</name>
+                    <type>SaturationVanGenuchten</type>
+                    <residual_liquid_saturation>0.18139535</residual_liquid_saturation>
+                    <residual_gas_saturation>0.0</residual_gas_saturation>
+                    <exponent>0.358974358974359</exponent>
+                    <p_b>2725</p_b>
+                </property>
+            </properties>
+        </medium>
+    </media>
+    <time_loop>
+        <processes>
+            <process ref="Hus">
+                <nonlinear_solver>NLS</nonlinear_solver>
+                <convergence_criterion>
+                    <type>DeltaX</type>
+                    <norm_type>NORM2</norm_type>
+                    <reltol>1e-8</reltol>
```

Comment:
Can you tighten the tolerance? I'd set the minimum which is still converging. This will reduce the spread of solutions due to different compiler/system setups.

Reply:
Can try, what would you consider a reasonable limit for this value?

Reply:
It is problem-dependent. I'd run it with 1e-16 first and observe the convergence behaviour. Then choose bigger value based on the output, s.t. the global iterations converge in reasonable number of steps (which, you guess it already, is problem-dependent ;) .

Reply:
For the sake of completeness: why starting with 1e-16?

Reply:
I think with a tol. of 1e-16 the equations are solved as accurately as possible on different CPU architectures / different operating systems / different compilers. I tried various thresholds and obtained different timesteps in which the timestamp (due to adaptive timestepping) is different from the reference file timestamp.

reltol   | timestep with different timestamp
---------|---------------------------------------------
1e-16 | 105
1e-14 | 525
1e-12 | 350

Reply:
1e-16 is close to machine epsilon for doubles.

Reply:
I once removed the timestamp from the file name, but also the actual simulation results differ quite a lot in these time steps.

Reply:
I am not sure how we can achieve that benchmarks with adaptive time-stepping on different architectures always converge in the same way. Maybe it is better to create the output files on envinf1 or in a docker container and only test the benchmarks in the same environment.

Reply:
> I am not sure how we can achieve that benchmarks with adaptive time-stepping on different architectures always converge in the same way. Maybe it is better to create the output files on envinf1 or in a docker container and only test the benchmarks in the same environment.

Or we have to set a defined time stepping size. Somewhat cumbersome...

Reply:
You may try `fixed_output_times ` for the adaptive time-stepping.  If that does not work due to file name difference,  you can dump the time step sizes by the  adaptive time-stepping on your machines, and then use the fixed time step with the dumped   time step sizes.

Reply:
Ok, thanks @wenqing for the hint. I will try this (using constant steps from one adaptive run) in the afternoon if there are no objections.

Reply:
I took a look at the time stepping and produced the following graph for different configurations of maximum time step size (60 and 120 s) and the non-linear iteration tolerance (1e-12 and 1e-9).
![Bild1](https://user-images.githubusercontent.com/6029954/82341869-414db080-99f1-11ea-9a27-1203a6a1344a.png)

I have the following concerns:
- You can see that all configurations show seemingly random drops in the time step size after the max has been reached and the simulation seemed to have stabilized (note that there is no change in boundary conditions, grid resolution nor parameter homogeneity). I cannot explain this, but it worries me somehow.
- Trying to find a version without these drops, I firstly decreased the tolerance (initially started with 1e-14), but couldn't find one without drops. I could decrease this further, but I guess @endJunction wouldn't be happy with this (and this is fine).
- Even more, for smaller maximum time step sizes, you can see that there are more frequent drops than for larger ones. Don't know if this is a good sign...

Now my question: how should we continue? I guess it is not a meaningful idea to decrease the tolerance further, nor does it seem rational to explicitely define all random drops (as they may, again, be different on other machines).

Reply:
@waltherm If you are happy with that tolerance, if you are confident that (most) errors will be caught when the code changes, you can choose any tolerance. Strict tolerances give less room for "error creep-in".

Reply:
@endJunction As I am already unconfident with the previous analysis (smaller time steps -> larger number of fluctuations), I don't know if I should be certain with this setup...